### ISSUE TITLE
After Training own synthetic dataset for few seconds -> Avg_dis: nan

### ISSUE THREAD

**Issue by marmas92** [j96w/DenseFusion#105]
Thank you again for your work.
While training with my own synthetic generated dataset formated like the linemod dataset, I get an Avg_dis: nan (see image). I checked twice if my metrics are in meters and they are. What am I doing wrong?

![image](https://user-images.githubusercontent.com/51153981/67636958-13c6db80-f8d6-11e9-913f-533bd7eaff39.png)

**Comment by j96w** [comment]
Hi, I remember I also had this problem when I switch from YCB to linemod but that was because of the meter metric. Usually, `nan` is because of some forwarding steps return inf value. I would guess it's because of the pose variation range in your training data is very big? Or could there be some data that is broken? Sorry, I can only provide my guess. Since currently DenseFusion has been tested on three datasets (YCB, linemod, NOCS), it should be quite robust.

**Comment by marmas92** [comment]
Hi, yes indeed my pose variation range is very big - is this a problem? Do I need to change the loss script?

**Comment by j96w** [comment]
How large is it? Does it satisfy the camera visible range?

**Comment by marmas92** [comment]
The objects are always in the camera visible range - what I am doing to generate the data is, I generate random objects with a random translation (in the camera visible range) and random rotation into a scene and then take picture with the ground trouth etc.

**Comment by j96w** [comment]
Sounds good. How about posting some generated data here?

**Comment by marmas92** [comment]
![0000](https://user-images.githubusercontent.com/51153981/67637701-6788f300-f8dd-11e9-8124-8b19e3388c18.png)  
RGB image
![0000](https://user-images.githubusercontent.com/51153981/67637704-6bb51080-f8dd-11e9-94d3-f82f3cb1de82.png) 
Segmentation mask
here would be the gt entry: 
0:
- cam_R_m2c: [-0.90200278, -0.42067076, 0.09709323, -0.02437935, -0.17490408, -0.9842836, 0.43104134, -0.89019361, 0.14750829]
  cam_t_m2c: [-0.09749632, 1.04666114, 1.57907629]
  obj_bb: [247, 164, 176, 175]
  obj_id: 1
here the info entry: 
0:
  cam_K: [888.888888888889, 320, 0, 0, 1000, 240, 0, 0, 1]
  depth_scale: 1

**Comment by j96w** [comment]
Ok! These synthetic data is good I think. How about lower the learning rate and take another try?

**Comment by marmas92** [comment]
It works, thanks! Could you tell my why it works now? ^^

**Comment by j96w** [comment]
To be honest, I don't know. But according to your runtime information, I found your training loss is jumping so I guess might lower the learning rate can help to stabilize it.

BTW, your synthetic data seems with high quality, would you mind let me know which simulation engine are you using? It's fine if it's a private tech.

**Comment by marmas92** [comment]
Ok, thanks!
Sure, I am using Unity to generate my data.

**Comment by j96w** [comment]
Thanks

**Comment by marmas92** [comment]
@j96w I have a quesetion regarding the  coordinate system your code is based on: is it left- or rigth-handed? Because I just realized the software (Unity) I use, uses the left-handed coordination system. So I probably will have wrong results if your code is based on the right-handed coordinate system...

**Comment by j96w** [comment]
It's right-handed. So you probably need to modify a little bit on the outputs.

**Comment by marmas92** [comment]
thnx!

**Comment by huckl3b3rry87** [comment]
I ran into the same issue using the FAT dataset, will try to lower the learning rate and report back.

I know that we are supposed to convert to meters [as per](https://github.com/j96w/DenseFusion#tips-for-your-own-dataset), and the FAT dataset is in cm, so I was dividing by 100 in the ``__getitem__()`` 

So, looking at the ``dataset.py`` scripts for[ Linemod](https://github.com/j96w/DenseFusion/blob/master/datasets/linemod/dataset.py#L183) and [YCB_Video_Dataset](https://github.com/j96w/DenseFusion/blob/master/datasets/ycb/dataset.py#L220), that would mean that these are in (mm) and (m), respectively. @j96w  is that correct?

**Comment by hatimwen** [comment]
> Hi, I remember I also had this problem when I switch from YCB to linemod but that was because of the meter metric. Usually, `nan` is because of some forwarding steps return inf value. I would guess it's because of the pose variation range in your training data is very big? Or could there be some data that is broken? Sorry, I can only provide my guess. Since currently DenseFusion has been tested on three datasets (YCB, linemod, NOCS), it should be quite robust.

Thanks for your work. And I notice that you've tested it on NOCS dataset, while I'm trying to test it on NOCS dataset with problems. So would you mind to make your codes about NOCS open? Thanks a lot again!

**Comment by GaoHuaZhang** [comment]
> I ran into the same issue using the FAT dataset, will try to lower the learning rate and report back.
> 
> I know that we are supposed to convert to meters [as per](https://github.com/j96w/DenseFusion#tips-for-your-own-dataset), and the FAT dataset is in cm, so I was dividing by 100 in the `__getitem__()`
> 
> So, looking at the `dataset.py` scripts for[ Linemod](https://github.com/j96w/DenseFusion/blob/master/datasets/linemod/dataset.py#L183) and [YCB_Video_Dataset](https://github.com/j96w/DenseFusion/blob/master/datasets/ycb/dataset.py#L220), that would mean that these are in (mm) and (m), respectively. @j96w is that correct?


I'm sorry to disturb you. but could you tell me which codes I should modify if my CAD model, that is. Ply file unit is m
